src/Jobs/AutoLiquidityManagementJob.cs

```diff
@@ -0,0 +1,323 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+using NBitcoin;
+using NodeGuard.Data.Models;
+using NodeGuard.Data.Repositories.Interfaces;
+using NodeGuard.Services;
+using Quartz;
+
+namespace NodeGuard.Jobs;
+
+/// <summary>
+/// Result of managing node liquidity operation
+/// </summary>
+public enum ManageNodeLiquidityResult
+{
+    Success,
+    BalanceBelowThreshold,
+    MaxSwapsInFlightReached,
+    BudgetExhausted,
+    ExcessBalanceBelowMinimum,
+    Error
+}
+
+/// <summary>
+/// Job for monitoring node balances and managing liquidity automatically based on node configuration.
+/// This includes swap outs, swap ins, and channel operations.
+/// </summary>
+[DisallowConcurrentExecution]
+public class AutoLiquidityManagementJob : IJob
+{
+    private readonly ILogger<AutoLiquidityManagementJob> _logger;
+    private readonly INodeRepository _nodeRepository;
+    private readonly ISwapOutRepository _swapOutRepository;
+    private readonly ISwapsService _swapsService;
+    private readonly ILightningService _lightningService;
+    private readonly IWalletRepository _walletRepository;
+    private readonly INBXplorerService _nbXplorerService;
+
+    public AutoLiquidityManagementJob(
+        ILogger<AutoLiquidityManagementJob> logger,
+        INodeRepository nodeRepository,
+        ISwapOutRepository swapOutRepository,
+        ISwapsService swapsService,
+        ILightningService lightningService,
+        IWalletRepository walletRepository,
+        INBXplorerService nbXplorerService)
+    {
+        _logger = logger;
+        _nodeRepository = nodeRepository;
+        _swapOutRepository = swapOutRepository;
+        _swapsService = swapsService;
+        _lightningService = lightningService;
+        _walletRepository = walletRepository;
+        _nbXplorerService = nbXplorerService;
+    }
+
+    public async Task Execute(IJobExecutionContext context)
+    {
+        _logger.LogInformation("Starting {JobName}...", nameof(AutoLiquidityManagementJob));
+
+        try
+        {
+            // Get all enabled nodes with auto liquidity management enabled
+            var nodes = await _nodeRepository.GetAllManagedByNodeGuard();
+            var eligibleNodes = nodes.Where(n => 
+                !n.IsNodeDisabled && 
+                n.AutoLiquidityManagementEnabled &&
+                n.FundsDestinationWalletId.HasValue).ToList();
+
+            _logger.LogInformation("Found {Count} nodes with automatic liquidity management enabled", eligibleNodes.Count);
+
+            foreach (var node in eligibleNodes)
```

Comment:
maybe it would be beneficial to do a thread per node here in case one node gets stuck

Reply:
yes, but a timeout in one node will prevent the other node from processing fast. It's not critical, though if eventually it times out

Reply:
Hmm swaps are not critical time-wise, its ok to be simpler imo